File: app/src/main/java/com/cn29/aac/ui/feedentrydetail/vm/FeedEntryDetailViewModel.java
package com.cn29.aac.ui.feedentrydetail.vm;


import com.cn29.aac.repo.feedentry.FeedEntry;
import com.cn29.aac.repo.feedentry.FeedEntryRepository;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

/**
 * Created by Charles Ng on 11/9/2017.
 */

public  class FeedEntryDetailViewModel extends ViewModel {

  private FeedEntryRepository feedEntryDBRepository;
  private LiveData<FeedEntry> feedEntry;
  private int uid;

  public FeedEntryDetailViewModel(FeedEntryRepository feedEntryRepository, int uid) {
    this.feedEntryDBRepository = feedEntryRepository;
    this.uid = uid;
    this.feedEntry = feedEntryDBRepository.findByUid(uid);

  }


  public LiveData<FeedEntry> getFeedEntry(int uid) {
    this.feedEntry = feedEntryDBRepository.findByUid(uid);
    return feedEntry;
  }

  public int update(FeedEntry feedEntry) {
  return feedEntryDBRepository.update(feedEntry);

  }


}

File: app/src/main/java/com/cn29/aac/ui/feedentrydetail/vm/FeedEntryDetailViewModel.kt
package com.cn29.aac.ui.feedentrydetail.vm

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.cn29.aac.repo.feedentry.FeedEntry
import com.cn29.aac.repo.feedentry.FeedEntryRepository

/**
 * Created by Charles Ng on 11/9/2017.
 */
class FeedEntryDetailViewModel(private val feedEntryDBRepository: FeedEntryRepository,
                               uid: Int) :
        ViewModel() {
    private var feedEntry: LiveData<FeedEntry?>?
    fun getFeedEntry(uid: Int): LiveData<FeedEntry?>? {
        feedEntry = feedEntryDBRepository.findByUid(uid)
        return feedEntry
    }

    fun update(feedEntry: FeedEntry?): Int {
        return feedEntryDBRepository.update(feedEntry)!!
    }

    init {
        feedEntry = feedEntryDBRepository.findByUid(uid)
    }
}